source/agora/node/Validator.d

```diff
@@ -413,6 +409,24 @@ public class Validator : FullNode, API
             this.config.validator.nomination_interval, &this.acceptBlock);
     }
 
+    /***************************************************************************
+
+        Returns an instance of a ValidatingLedger.
+
+        Test-suites can inject different behaviour to enable testing.
+
+        Returns:
+            An instance of a `ValidatingLedger`
+
+    ***************************************************************************/
+    protected ValidatingLedger makeValidatingLedger ()
```

Comment:
Newline in between. And this should probably be `makeLedger`.
The whole Ledger instantiation is still pretty bad, because the `FullNode` has a if to avoid a needless instantiation.
Can the `FullNode` use this too ?

Reply:
Yes good idea, I'll update.

Reply:
Updated by introducing ILedger as an abstract class with the `Ffullnode` implementation. In `Validator` the 'makeLedger' is overridden to create a `ValidatingLedger` which is an implementation of `ILedger`.